src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
```

Comment:
I don't agree with you, because my code already has the ArrayList() constructor, located on 10-12 lines. We need the second constructor ArrayList(int initialCapacity).

https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/util/ArrayList.html#constructor-summary